okhttp/src/main/java/okhttp3/internal/http2/Http2Codec.java

```diff
@@ -90,13 +90,17 @@
   final StreamAllocation streamAllocation;
   private final Http2Connection connection;
   private Http2Stream stream;
+  private final Protocol protocol;
 
   public Http2Codec(OkHttpClient client, Interceptor.Chain chain, StreamAllocation streamAllocation,
       Http2Connection connection) {
     this.client = client;
     this.chain = chain;
     this.streamAllocation = streamAllocation;
     this.connection = connection;
+
+    // cache this so we don't do linear scans on every call
```

Comment:
this comment is misleading. “linear scans” is on a list of a very small constant size.

Reply:
(I think I’d just omit the comment)